pkg/convert/pprof/streaming/parser_streaming.go

```diff
@@ -95,7 +101,7 @@ func (p *VTStreamingParser) ParsePprof(ctx context.Context, startTime, endTime t
 		if err != nil {
 			err = fmt.Errorf("failed to create pprof profile zip reader: %w", err)
 		} else {
-			buf := PPROFBufPool.Get()
+			buf := PPROFBufPool.Get() //todo consider decompression on arena
```

Comment:
Well, if you need my opinion, I think pools and arenas pursue different goals:
 1. Pools allow reusing already allocated objects, reducing number of allocations.
 2. Arenas allow to allocate objects from a contiguous region of memory and free them all at once, reducing garbage collection overhead caused by large number of objects on heap. Objects allocated on arena can't/shouldn't be reused.

Therefore, I doubt moving the buffer to arena will improve preformance

Reply:
So, I dont know if will help or not, I just wanted to try it :)

My thought process is that I believe pool is cleaned on each gc cycle so allocations still happen sometime. At the same time arena chunk is 8m and if we don't use all of it during parsing, which I hope we dont(except maybe we do for cumulative profiles), we could use some of it for decompression. Also there is a huge chance I will not be able to measure the deference at all.

Reply:
I find this interesting – it seems that a object bigger than 2MB (given the chunk size) will be allocated on heap (I'm using 1.20rc3): 
```go
	// userArenaChunkMaxAllocBytes is the maximum size of an object that can
	// be allocated from an arena. This number is chosen to cap worst-case
	// fragmentation of user arenas to 25%. Larger allocations are redirected
	// to the heap.
	userArenaChunkMaxAllocBytes = userArenaChunkBytes / 4
```
```go
	if size > userArenaChunkMaxAllocBytes {
		// Redirect allocations that don't fit into a chunk well directly
		// from the heap.
		if cap >= 0 {
			return newarray(typ, cap)
		}
		return newobject(typ)
	}
```